PLATUI-629

ejayaraman opened the pull request:
Upgrade to SBT to 1.3.13. Scala 2.12
Update dependencies required to use SBT 1.x.x and Scala 2.12

ejayaraman committed:
PLATUI-629: Update SBT to 1.3.13 and Scala to 2.12.12
Introduced build.sbt
Updated dependencies version to support SBT and Scala upgraded.

ejayaraman committed:
PLATUI-629: updated showClasspath task name

nigelhp commented on build.sbt:
```diff
@@ -0,0 +1,40 @@
+val scalaOptions = Seq(
```
These should already be configured via `sbt-auto-build` which in turn uses `sbt-settings`.
The only one I don't see is `-language` but I'm not sure if a wildcard actually works there (it would seem a little dangerous if it does).  The docs [scala compiler options](https://docs.scala-lang.org/overviews/compiler-options/index.html) do not mention wildcard support - but this could always be an oversight.

ejayaraman replied:
Removed all existing scalac options. `-language` is the only setting that is not included in `sbt-auto-build`. But it appears too dangerous to include as a wildcard. So I have removed that setting and introduced `-feature` which will surface detailed warnings when an advanced featured is used without relevant settings/imports.

nigelhp commented on project/Dependencies.scala:
```diff
@@ -3,13 +3,13 @@ import sbt._
 object Dependencies {
 
   object Compile {
```
In our template projects for services the convention is for the dependencies file to expose the dependencies as ready-made sequences e.g.

```
object AppDependencies {
  val compile = Seq(...)
  val test = Seq(...)
}
```

The `build.sbt` file then simply contains:
```
libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test,
```

ejayaraman replied:
Marked all dependencies as `Test` and converted into a `Seq` as suggested above.

ejayaraman committed:
PLATUI-629: Review comment changes.
- Removed scalacOptions as this provided by sbt-auto-build.
- Restructured Dependencies and marked them as Test dependencies.
- Updated showClassPath task to show fullClasspath in Test

ejayaraman committed:
PLATUI-629: Fixed postfixOps warning by introduce relevant import.  Introduced as a result of removing Scalac language settings.
 Introduce `-feature` scalac setting as this provides detailed feature warning messages.
